### DISCUSSION TITLE
Monitor subdirectories for changed files

### DISCUSSION THREAD

**Discussion by m-guggenmos** [quarto-dev/quarto-cli#6318] [Q&A] [answered]
### Description

Quarto newbie question. I created a new `website` project and was pleased to see that the html preview is automatically refreshed when I change qmd files in the top-level root directory of my project (using `quarto preview`). However, subdirectories do not seem to be monitored for changes in qmd files. Can I configure this somehow?

I'm using VS Code with the Quarto extension.

**Comment by mcanouil** [comment]
Quarto only watches needed resources, it's not really about directories.
Without knowing your exact setup, it's impossible to fully answer.

**Comment by m-guggenmos** [comment]
Ok, I found a way for now. The quarto extension has a hard-coded command line argument `--no-watch-inputs` for `quarto preview`. I commented the line `cmd.push("--no-watch-inputs")` in `~/.vscode/extensions/quarto.quarto-1.90.0/out/main.js` and it now seems to listen to file changes also in subdirectories.

**Reply by cderv** [reply]
@jjallaire FYI 

Should we try to make this an option in Quarto VSCODE extension if this has some use case ? 

Or maybe we did deactivate it for some specific reason ? 

https://github.com/quarto-dev/quarto/blob/a92906bbd004e3d8abc39b5f59346ca2a091d1e8/apps/vscode/src/providers/preview/preview.ts#L461

**Reply by mcanouil** [reply]
Probably for the same reason as the "render on save" feature is deactivated by default, but I agree that an option is advisable.

**Reply by m-guggenmos** [reply]
What confused me initially is the fact that there seems to be a `quarto.yml` option to control this behavior:

https://quarto.org/docs/reference/projects/websites.html#preview

According to the docs `watch-inputs` is `true` by default.

**Reply by mcanouil** [reply]
Yes but it's `_quarto.yml`.
(I don't know if the VSCode extension respects that but it should)

**Comment by jjallaire** [comment, marked as answer]
The reason we force `--no-watch-inputs` is that the VS Code extension does explicit rendering of documents (either vis Render or Render on Save). If the preview server were also auto-rendering on changes then you would get a double render (which for some documents could be very expensive). That said, we may have subsequently put in protection against double renders so it might appear to work w/o troubles when you comment that out. 

If you use the `render-on-save` option in VS Code then you more or less get the behavior you want. The only thing it doesn't do is automatically detect changes to documents that are **not** currently active in the preview window. You could get this to work by just running `quarto preview` from a terminal then just editing/saving as normal in VS Code.

**Reply by m-guggenmos** [reply]
Thanks a lot @jjallaire , that is very helpful!
